chips/nrf51/src/rtc.rs

```diff
@@ -94,23 +105,33 @@ impl Rtc {
             cb.fired();
         });
     }
+
+    pub fn set_client(&self, client: &'static AlarmClient) {
+        self.callback.set(Some(client));
+    }
 }
 
 impl Alarm for Rtc {
-    type Frequency = Freq16KHz;
+    type Frequency = Freq32KHz;
 
     fn now(&self) -> u32 {
         rtc1().counter.get()
     }
 
     fn disable_alarm(&self) {
-        self.stop();
+        //Rather than stopping the timer itself, we just stop listening for it
+        //If we were to turn it off entirely, it would add a large amount of overhead each tick
+        rtc1().cc[0].set(0);
+        nvic::disable(NvicIdx::RTC1);
+        rtc1().intenclr.set(COMPARE0_EVENT);
     }
 
     fn set_alarm(&self, tics: u32) {
+        //Similarly to the disable function, here we don't restart the timer
+        //Instead, we just listen for it again
         rtc1().cc[0].set(tics);
         rtc1().intenset.set(COMPARE0_EVENT);
-        self.start();
+        nvic::clear_pending(NvicIdx::RTC1);
```

Comment:
In general we shouldn't have to explicitly clear the pending NVIC, it should be cleared by us handling the event (this seems to be in several places)